src/StyleSheet.js

```diff
@@ -30,22 +29,26 @@ export default class StyleSheet {
     this.rules = Object.create(null)
     // Used to ensure correct rules order.
     this.rulesIndex = []
-    this.classes = Object.create(null)
+    this.classes = this.options.classes = Object.create(null)
```

Comment:
As an implementation detail I think it is okay. I understand what you are saying about the additional overhead.

Quick Q, when you say:

> Oh, can't do the last one, classes is a hash of all classes for the sheet, it needs to be accessed from all RulesContainer instances.

What do you mean by **all** instances? I thought that StyleSheet -> RulesContainer was a 1:1 relationship.